src/Data/List/Ext.agda

```diff
@@ -35,3 +40,23 @@ insert x (y ∷ ys) = (x ∷ y ∷ ys) ∷ map (y ∷_) (insert x ys)
 subpermutations : List A → List (List A)
 subpermutations [] = [] ∷ []
 subpermutations (x ∷ xs) = concatMap (insert x) (subpermutations xs) ++ subpermutations xs
+
+∈-map⁻ : {f : A → B} {b : B} {l : List A} → b ∈ map f l → ∃[ a ](b ≡ f a × a ∈ l)
+∈-map⁻ {l = _ ∷ _} (here _≡_.refl) = _ , _≡_.refl , here _≡_.refl
+∈-map⁻ {l = _ ∷ _} (there h) = proj₁ (∈-map⁻ h) , proj₁ (proj₂ (∈-map⁻ h)) , there (proj₂ (proj₂ (∈-map⁻ h)))
+
+∈-map⁺ : {f : A → B} {b : B} {l : List A} → ∃[ a ](b ≡ f a × a ∈ l) → b ∈ map f l
+∈-map⁺ (._ , b≡fa , here _≡_.refl) = here b≡fa
+∈-map⁺ (a , b≡fa , there a∈l) = there (∈-map⁺ (a , b≡fa , a∈l))
+
+∈-map : {f : A → B} {b : B} {l : List A} → (∃[ a ] (b ≡ f a × a ∈ l)) ⇔ b ∈ map f l
+∈-map {l = []} = mk⇔ (λ ()) (λ ())
+∈-map {l = _ ∷ _} = mk⇔ ∈-map⁺ ∈-map⁻
```

Comment:
They're not quite the same.  The first two differ in signatures and `∈-map` inhabits `Equivalence`, whereas `map-∈↔` (in `Data.List.Membership.Propositional.Properties`) inhabits `Inverse`. I can make adjustments and replace `∈-map⁻` and `∈-map⁺` with those from `Data.List.Membership.Propositional.Properties`, but I think we should `∈-map` around.  (Otherwise, I'll need to figure out how to convert from `Inverse` to `Equivalence`.)

Reply:
We're already doing the conversion from `Inverse` to `Equivalence` in lots of places. It's done with `↔⇒` from `Function.Related.Propositional`, which is usually available as `R.↔⇒`.

Reply:
By "lots" I think you mean "two" 😉 ...but I'll give it a try.

The only occurrences of they symbol `↔⇒` I could find are shown below.

```
File: Axiom/Set/List.agda
39:61:    (∃[ a ] b ≡ f a × a ∈ˡ X) ∼⟨ ∃-cong′ (I.↔⇒⇔ (×-comm _ _)) ⟩
40:48:    (∃[ a ] a ∈ˡ X × b ≡ f a) ⤖⟨ I.↔⇒⤖ (map-∈↔ f) ⟩

File: Ledger/Gov.lagda
28:51:open import Function.Related.Propositional using (↔⇒)
163:46:∃?-connecting-subperm L = from (map′⇔ (↔⇒ Any↔)) (any?-connecting-subperm L)
4 matches
```